src/selectors/nodes.js

```diff
@@ -113,10 +109,26 @@ export const getNodeData = createSelector(
         disabled: nodeDisabled[id],
         disabled_node: Boolean(nodeDisabledNode[id]),
         disabled_tag: nodeDisabledTag[id],
-        disabled_view: nodeDisabledView[id]
+        disabled_type: Boolean(typeDisabled[nodeType[id]])
       }))
 );
 
+/**
+ * Returns formatted nodes grouped by type
+ */
+export const getGroupedNodes = createSelector(
+  [getNodeData],
+  nodes =>
+    nodes.reduce(function(obj, item) {
+      const key = item.type;
+      if (!obj.hasOwnProperty(key)) {
```

Comment:
Does this need hasOwnProperty or can it simple be

```
if (!obj[key])
```

Reply:
It probably could, but this is safer as it will still create keys for falsey values like empty strings

Reply:
Why is it safer though? I could see it in use if it was a class / function that had properties inside it rather than on the prototype but afaik nodes is an array of objects?:

```
function A() {
   this.x = "I'm an own property";
}

A.prototype.y = "I'm not an own property";

var instance = new A();
var xIsOwnProperty = instance.hasOwnProperty("x"); // true
var yIsOwnProperty = instance.hasOwnProperty("y"); // false
```

```
var obj = {}
if (!obj['']) console.log('hello') // logs "hello"
if (!obj[null]) console.log('hello') // logs "hello"
if (!obj[undefined]) console.log('hello') // logs "hello"
```

It works either way but just wondering 🙂

Reply:
I meant if the property value itself were falsey.

e.g.
```
const obj = {
 foo: 0
};
console.log(!obj['foo']); // true
console.log(!obj.hasOwnProperty('foo')); // false
```

I figured it didn't matter too much either way, but this way is slightly more explicit and potentially safer as it's only checking for whether the key is present. I don't feel strongly about it but I'm pretty sure it doesn't matter?

Reply:
Cool happy to leave as it is then